Compilation issue

Hi!

```bash
cargo new thing
cd thing
# This is OK
cargo build

cargo add tokio-tower

# Fails
cargo build
```

Error output:

```
error[E0425]: cannot find function `spawn` in crate `tokio`
   --> MYPATH\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-tower-0.4.0\src\multiplex\client.rs:196:16
    |
196 |         tokio::spawn({
    |                ^^^^^ not found in `tokio`
    |
help: consider importing this function
    |
1   | use std::thread::spawn;
    |

error[E0425]: cannot find function `spawn` in crate `tokio`
   --> MYPATH\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-tower-0.4.0\src\pipeline\client.rs:173:16
    |
173 |         tokio::spawn({
    |                ^^^^^ not found in `tokio`
    |
help: consider importing this function
    |
1   | use std::thread::spawn;
    |

error: aborting due to 2 previous errors
```

More info:
```
cargo -V
cargo 1.50.0-nightly (d274fcf86 2020-12-07)
```

But same on a `+stable` build.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@ pin-project = "1.0"
 tokio = { version = "1.0", features = [ "full" ] }
 serde = "1.0"
 serde_derive = "1.0"
-async-bincode = { version = "0.5", git = "https://github.com/taiki-e/async-bincode", branch = "deps" }
+async-bincode = "0.6"
 slab = "0.4"
 tokio-test = "0.4"
 tower-test = "0.4"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,7 +26,7 @@ tower = { version = "0.4", features = ["load"] }
 futures-util = { version = "0.3", features = [ "sink" ] }
 futures-core = "0.3"
 futures-sink = "0.3"
-tokio = { version = "1.0", features = [ "sync" ] }
+tokio = { version = "1.0", features = [ "sync", "rt" ] }
 crossbeam = "0.8"
 tracing = "0.1.2"
 pin-project = "1.0"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,20 +22,20 @@ default = []
 
 [dependencies]
 tower-service = "0.3"
-tower-load = "0.3"
+tower = { version = "0.4", features = ["load"] }
 futures-util = { version = "0.3", features = [ "sink" ] }
 futures-core = "0.3"
 futures-sink = "0.3"
-tokio = { version = "0.2", features = [ "sync" ] }
-crossbeam = "0.7"
+tokio = { version = "1.0", features = [ "sync" ] }
+crossbeam = "0.8"
 tracing = "0.1.2"
-pin-project = "0.4.0"
+pin-project = "1.0"
 
 [dev-dependencies]
-tokio = { version = "0.2", features = [ "macros", "tcp", "dns", "rt-core" ] }
+tokio = { version = "1.0", features = [ "full" ] }
 serde = "1.0"
 serde_derive = "1.0"
-async-bincode = "0.5"
+async-bincode = { version = "0.5", git = "https://github.com/taiki-e/async-bincode", branch = "deps" }
 slab = "0.4"
-tokio-test = "0.2"
-tower-test = "0.3"
+tokio-test = "0.4"
+tower-test = "0.4"

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,7 +1,7 @@
 jobs:
  - template: default.yml@templates
    parameters:
-     minrust: 1.39.0
+     minrust: 1.45.2
 
 resources:
   repositories:

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -40,7 +40,6 @@
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
 //!
-/* TODO: https://github.com/tokio-rs/tokio/pull/3105
 //! # Example
 //! ```rust
 //! # use std::pin::Pin;
@@ -63,20 +62,20 @@
 //! /// In this echo server example, requests and responses are both of type `T`, but for "real"
 //! /// services, the two types are usually different.
 //! struct ChannelTransport<T> {
-//!     rcv: mpsc::Receiver<T>,
-//!     snd: mpsc::Sender<T>,
+//!     rcv: mpsc::UnboundedReceiver<T>,
+//!     snd: mpsc::UnboundedSender<T>,
 //! }
 //!
 //! impl<T: Debug> futures_sink::Sink<T> for ChannelTransport<T> {
 //!     type Error = StdError;
 //!
 //!     fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
-//!         self.snd.poll_ready(cx).map_err(|e| e.into())
+//!         Poll::Ready(Ok(()))
 //!     }
 //!
 //!     fn start_send(mut self: Pin<&mut Self>, item: T) -> Result<(), Self::Error> {
-//!         // unwrap ok because of poll_ready()
-//!         self.snd.try_send(item).unwrap();
+//!         // use map_err because `T` contained in `mpsc::SendError` may not be `Send + Sync`.
+//!         self.snd.send(item).map_err(|e| e.to_string())?;
 //!         Ok(())
 //!     }
 //!
@@ -117,8 +116,8 @@
 //!
 //! #[tokio::main]
 //! async fn main() {
-//!     let (s1, r1) = mpsc::channel(2);
-//!     let (s2, r2) = mpsc::channel(2);
+//!     let (s1, r1) = mpsc::unbounded_channel();
+//!     let (s2, r2) = mpsc::unbounded_channel();
 //!     let pair1 = ChannelTransport{snd: s1, rcv: r2};
 //!     let pair2 = ChannelTransport{snd: s2, rcv: r1};
 //!
@@ -134,7 +133,6 @@
 //! }
 //!
 //! ```
-*/
 #![warn(
     missing_docs,
     missing_debug_implementations,

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -40,6 +40,7 @@
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
 //!
+/* TODO: https://github.com/tokio-rs/tokio/pull/3105
 //! # Example
 //! ```rust
 //! # use std::pin::Pin;
@@ -133,6 +134,7 @@
 //! }
 //!
 //! ```
+*/
 #![warn(
     missing_docs,
     missing_debug_implementations,

--- a/src/multiplex/client.rs
+++ b/src/multiplex/client.rs
@@ -105,7 +105,7 @@ where
     }
 }
 
-impl<NT, Request> tower_load::Load for Maker<NT, Request> {
+impl<NT, Request> tower::load::Load for Maker<NT, Request> {
     type Metric = u8;
 
     fn load(&self) -> Self::Metric {
@@ -417,7 +417,7 @@ where
     }
 }
 
-impl<T, E, Request> tower_load::Load for Client<T, E, Request>
+impl<T, E, Request> tower::load::Load for Client<T, E, Request>
 where
     T: Sink<Request> + TryStream,
 {

--- a/src/pipeline/client.rs
+++ b/src/pipeline/client.rs
@@ -84,7 +84,7 @@ where
     }
 }
 
-impl<NT, Request> tower_load::Load for Maker<NT, Request> {
+impl<NT, Request> tower::load::Load for Maker<NT, Request> {
     type Metric = u8;
 
     fn load(&self) -> Self::Metric {
@@ -377,7 +377,7 @@ where
     }
 }
 
-impl<T, E, Request> tower_load::Load for Client<T, E, Request>
+impl<T, E, Request> tower::load::Load for Client<T, E, Request>
 where
     T: Sink<Request> + TryStream,
 {

--- a/tests/multiplex/mod.rs
+++ b/tests/multiplex/mod.rs
@@ -26,7 +26,7 @@ impl TagStore<Request, Response> for SlabStore {
 
 #[tokio::test]
 async fn integration() {
-    let mut rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = rx.local_addr().unwrap();
 
     // connect
@@ -55,7 +55,7 @@ async fn integration() {
 
 #[tokio::test]
 async fn racing_close() {
-    let mut rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = rx.local_addr().unwrap();
 
     // connect

--- a/tests/pipeline/client.rs
+++ b/tests/pipeline/client.rs
@@ -7,7 +7,7 @@ use tower_service::Service;
 
 #[tokio::test]
 async fn it_works() {
-    let mut rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = rx.local_addr().unwrap();
 
     // connect

--- a/tests/pipeline/mod.rs
+++ b/tests/pipeline/mod.rs
@@ -10,7 +10,7 @@ mod client;
 
 #[tokio::test]
 async fn integration() {
-    let mut rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = rx.local_addr().unwrap();
 
     // connect
@@ -38,7 +38,7 @@ async fn integration() {
 
 #[tokio::test]
 async fn racing_close() {
-    let mut rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let rx = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = rx.local_addr().unwrap();
 
     // connect

